miniml/compiler/compile.scm

```diff
@@ -632,6 +643,9 @@
                             (if (char=? (peek-char) #\=)
                                 (begin (read-char) (make-lexical-token 'COLONEQ location #f))
                                 (make-lexical-token 'COLON location #f))))
+        ((char=? c #\%) (if (try-read 0 "exit")
```

Comment:
There is already a case for `%` below, since `%` is allowed as part of an operator. Couldn't we lex `[%` specially instead, and add a rule `LET PERCENT LIDENT ...` and then check the identifier is `exit`? This would avoid having to treat `%` specially (apart from adding it as a special case in the `operator-kw` table, and would even be easier to extend to other extension constructions.

Reply:
Actually I started this way and I was sort of discouraged by having to raise syntax errors from the semantic action (which parsing experts tell me is bad form). I agree that it is more flexible, so I will go back to this approach.

Reply:
If I understand correctly, "%" is currently lexed as an INFIXOP3. To give it a special status in the grammar I need to turn it into its own token (so PERCENT stays), as you are already doing for certain operators (COLONEQ, EQ, MINUS, STAR). This is how the OCaml grammar does it. It also defines a BRACKETPERCENT, but as far as I can tell this is only used to prevent spaces between `[` and `%` in extensions, and I don't really care about this so I would rather go without to keep the lexer simpler.

Reply:
I tried to implement something along these lines, let me know what you think.